### PR TITLE
Csi 4745/automatic port configuration sotrage manager implantation

### PR DESCRIPTION
**Pull request by matancarmeli7**

### CODE REVIEW

File: controllers/servers/host_definer/types.py [hunk 1]
```diff
@@ -5,14 +5,18 @@
...
+    node_id_from_csi_node: str = ''
     array_connection_info: ArrayConnectionInfo = ArrayConnectionInfo(array_addresses='', user='', password='')
 
 
 @dataclass
 class DefineHostResponse:
     error_message: str = ''
+    connectivity_type: str = ''
+    ports: list = field(default_factory=list)
+    node_name_on_storage: str = ''
```
Comment (ArbelNathan) [inline controllers/servers/host_definer/types.py:19]: is there a real need to define a default value? are they all can be not initiated?

File: controllers/servers/host_definer/storage_manager/host_definer_server.py [hunk 2]
```diff
@@ -61,9 +66,62 @@ def _get_host_name(self, initiators, array_mediator):
         logger.debug("host found : {}".format(found_host_name))
         return found_host_name
 
+    def _update_host_ports(self, request, host, array_mediator):
```
Comment (ArbelNathan) [inline controllers/servers/host_definer/storage_manager/host_definer_server.py:69]: (comment for most new private methods)
can we move it somewhere else? (i.e. utils, helper...)

File: controllers/servers/host_definer/types.py (hunk 1 above)
Comment (matancarmeli7) [reply controllers/servers/host_definer/types.py:19]: can I open a ticket for it and for all the other dataclasses in the hostDefinition?

File: controllers/servers/host_definer/storage_manager/host_definer_server.py (hunk 2 above)
Comment (matancarmeli7) [reply controllers/servers/host_definer/storage_manager/host_definer_server.py:69]: why?

Comment (matancarmeli7) [reply controllers/servers/host_definer/storage_manager/host_definer_server.py:69]: we can do a brainstorming on which functions to import out to different classes if you want, as we did with the Kubernetes manager in the hostDefiner

File: controllers/servers/host_definer/types.py (hunk 1 above)
Comment (ArbelNathan) [reply controllers/servers/host_definer/types.py:19]: sure, put here ticket num

File: controllers/servers/host_definer/storage_manager/host_definer_server.py (hunk 2 above)
Comment (ArbelNathan) [reply controllers/servers/host_definer/storage_manager/host_definer_server.py:69]: just not to overload the class.. can you open a ticket for a discussion?

Comment (matancarmeli7) [reply controllers/servers/host_definer/storage_manager/host_definer_server.py:69]: ok

Comment (matancarmeli7) [reply controllers/servers/host_definer/storage_manager/host_definer_server.py:69]: CSI-4867

File: controllers/servers/host_definer/types.py (hunk 1 above)
Comment (matancarmeli7) [reply controllers/servers/host_definer/types.py:19]: CSI-4868

**Review by ArbelNathan — DISMISSED** [review]
LGTM GJ!

**Review by ELENAGER — DISMISSED** [review]
LGTM

**Review by ELENAGER — APPROVED** [review]
LGTM

**Pull request merged**